internal/core/services/discovery.go

```
@@ -0,0 +1,166 @@
+		Body: protocol.DiscoverFeatureDiscloseMessageBody{
+			Disclosures: disclosures,
+		},
+		From: from,
+		To:   to,
+	}, nil
+}
+
+func (d *discovery) handleAccept(_ context.Context) ([]protocol.DiscoverFeatureDisclosure, error) {
+	disclosures := []protocol.DiscoverFeatureDisclosure{}
+
+	profiles := d.packerManager.GetSupportedProfiles()
+	for _, profile := range profiles {
+		disclosures = append(disclosures, protocol.DiscoverFeatureDisclosure{
+			FeatureType: protocol.DiscoveryProtocolFeatureTypeAccept,
+			ID:          profile,
+		})
+	}
+	return disclosures, nil
+}
+
+func (d *discovery) handleProtocol(_ context.Context) []protocol.DiscoverFeatureDisclosure {
+	disclosures := []protocol.DiscoverFeatureDisclosure{}
+	for _, protocolMessage := range d.supportedProtocols {
+		disclosures = append(disclosures, protocol.DiscoverFeatureDisclosure{
+			FeatureType: protocol.DiscoveryProtocolFeatureTypeProtocol,
+			ID:          string(protocolMessage),
+		})
+	}
+	return disclosures
+}
+
+func (d *discovery) handleGoalCode(_ context.Context) []protocol.DiscoverFeatureDisclosure {
+	disclosures := []protocol.DiscoverFeatureDisclosure{}
+	return disclosures
+}
+
+func (d *discovery) handleHeader(_ context.Context) []protocol.DiscoverFeatureDisclosure {
+	headers := []string{
+		"id",
+		"typ",
+		"type",
+		"thid",
+		"body",
+		"from",
+		"to",
+		"created_time",
+		"expires_time",
+	}
+
+	disclosures := []protocol.DiscoverFeatureDisclosure{}
+
+	for _, header := range headers {
+		disclosures = append(disclosures, protocol.DiscoverFeatureDisclosure{
+			FeatureType: protocol.DiscoveryProtocolFeatureTypeHeader,
+			ID:          header,
+		})
+	}
+
+	return disclosures
+}
+
+func wildcardToRegExp(match string) *regexp.Regexp {
+	// Escape special regex characters and replace `*` with `.*`
+	regexPattern := regexp.QuoteMeta(match)
+	regexPattern = strings.ReplaceAll(regexPattern, "\\*", ".*")
+	regExp, _ := regexp.Compile("^" + regexPattern + "$")
```

Is it safe to ignore this error? Are we sure that regexPattern is valid?